src/metabase/server/middleware/session.clj

```diff
@@ -359,21 +359,46 @@
 ;;; |                                              reset-cookie-timeout                                             |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+(defn- check-session-timeout
+  "Returns nil if the [[session-timeout]] value is valid. Otherwise returns an exception."
+  [timeout]
+  (when (some? timeout)
+    (let [{:keys [unit amount]} timeout
+          units-in-24-hours (case unit
+                              "seconds" (* 60 60 24)
+                              "minutes" (* 60 24)
+                              "hours"   24)
+          units-in-100-years (* units-in-24-hours 365.25 100)]
+      (cond
+        (not (pos? amount))
+        (Exception. (tru "Session timeout amount must be positive."))
+        (>= amount units-in-100-years)
+        (Exception. (tru "Session timeout must be less than 100 years."))))))
+
 (defsetting session-timeout
   ;; Should be in the form {:amount 60 :unit "minutes"} where the unit is one of "seconds", "minutes" or "hours".
   ;; The amount is nillable.
   (deferred-tru "Time before inactive users are logged out. By default, sessions last indefinitely.")
-  :type       :json
-  :default    nil)
+  :type    :json
+  :default nil
+  :getter  (fn []
+             (let [value (setting/get-value-of-type :json :session-timeout)]
+               (if-let [exception (check-session-timeout value)]
+                 (println (str "WARNING: " (ex-message exception)))
```

Comment:
Don't use `println` here since people can't configure it

We should probably add some sort of Kondo warning for that

Reply:
At any rate `WARNING` isn't i18n'ed

Reply:
For log messages, it should be i18n'ed with `trs` rather than `tru` -- if the system locale is English but some user has their locale set to French it's going to be confusing for the admin when a bunch of error messages in French start popping up in the logs. You might want to have a separate message for the warning or do this in some other way